nimbleparse/src/main.rs

```diff
@@ -346,24 +346,156 @@ fn main() {
+                                    Err(e) => {
+                                        eprintln!(
+                                            "Error globbing %grmtools 'test_files' {}",
+                                            e.to_string()
+                                        );
+                                        process::exit(1);
+                                    }
+                                }
+                            }
+                            ret
+                        } else {
+                            eprintln!(
+                                "Error globbing %grmtools 'test_files', {}",
+                                paths.unwrap_err().to_string()
+                            );
+                            process::exit(1);
+                        }
+                    }
+                    Value::Setting(Setting::Num(_, loc)) => {
+                        eprintln!(
+                            "Expected 'test_files: \"glob\"' found bool in '%grmtools' from '{:?}'",
+                            loc
+                        );
+                        process::exit(1);
+                    }
+                    Value::Setting(Setting::Unitary(Namespaced {
+                        namespace: _,
+                        member: (member, memb_loc),
+                    })) => {
+                        eprintln!("Expected 'test_files: \"glob\"' found {member} in '%grmtools' from '{:?}'", memb_loc);
+                        process::exit(1);
+                    }
+                    Value::Setting(Setting::Constructor {
+                        ctor:
+                            Namespaced {
+                                namespace: _,
+                                member: (ctor_memb, ctor_memb_loc),
+                            },
+                        arg:
+                            Namespaced {
+                                namespace: _,
+                                member: (arg_memb, _),
+                            },
+                    }) => {
+                        eprintln!("Expected 'test_files: \"glob\"' found constuctor '{:?}::{:?}' in '%grmtools' from: {:?}", ctor_memb, arg_memb, ctor_memb_loc);
+                        process::exit(1);
+                    }
+                }
+            } else {
+                eprintln!("Missing <input file> argument");
+                process::exit(1);
+            }
+        } else {
+            input_paths
+                .iter()
+                .map(|path| PathBuf::from(path))
+                .collect::<Vec<PathBuf>>()
+        };
+        if input_paths.is_empty() {
+            eprintln!("Missing <input file> argument");
+            process::exit(1);
+        }
+        for input_path in input_paths {
+            let input = read_file(&input_path);
+            let lexer = lexerdef.lexer(&input);
+            let pb = RTParserBuilder::new(&grm, &stable).recoverer(recoverykind);
```

Comment:
Should try to pull the contruction of `RTParserBuilder` here up out of the loop.
We really shouldn't need to rebuild it for every input file, seems like an oversight.

Reply:
Should be fixed in [bd406bb](https://github.com/softdevteam/grmtools/pull/559/commits/bd406bb7d1a34a8555f84438f67510586bb867cc), also there was another issue with errors being ascribed to the wrong source file (the grammar.y rather than the input.txt).